### ISSUE TITLE
看了您的报告之后，有些代码上的问题需要向您请教

### ISSUE THREAD

**Issue by 1048846280** [jianganbai/AEGAN-AD#3]
尊敬的姜安柏博士：
您好！
非常感谢您的在机器音频异常检测工作上的贡献，听完您的报告，我尝试复现代码的dcase22中bearing部分，过程中遇到了几个问题想请教一下您。
1. dcase22共提供了3部分数据集，我将Development dataset解压到了dev_data中，Additional training dataset和Evaluation dataset解压到了eval_data中，不知道这样是不是正确？
2. 我使用了您提供的pretrained dicts进行测试，测试的输出结果如下：   
 
                    ========= Test Machine Type: bearing =========
                    ============== TEST DATASET GENERATOR ==============
                    ============== TRAIN DATASET GENERATOR ==============
                    => Recorded best hmean: 0.7603
                    => Detection on dev test set
                    100%|██████████| 600/600 [00:51<00:00, 11.60it/s]
                    => Best metric: G_z_1_sum; Best hmean: 0.7075

 其中的best hmean 只有0.7075，是不是在第一个问题中数据集没有设置好，还是其他的原因？

感谢您在百忙之中阅读我的邮件，谢谢！
如果您能解答我的疑惑，我将不胜感激。
 
祝您生活愉快，科研顺利！

**Comment by jianganbai** [comment]
您好！

感谢对我们工作的关注。

效果偏低可能是因为没有用pretrain文件夹中的yaml配置文件，替换原始的config.yaml文件。

原始的config.yaml只在dev集上训练，bearing预训练参数是在dev+eval两个数据集上训练的。

输入样本会根据数据集的均值和方差进行归一化，使用错误的config.yaml，会只使用dev集的统计量进行归一化，导致结果变差。

祝好！

**Comment by 1048846280** [comment]
非常感谢您百忙之中的回复！

我们确定使用了bearing_cfg.yaml中的参数设置，只在dataset_dir处做了改变。同时又观察了spec\deval\bearing文件夹中生成的train_scale_mel_128_2048_512_1.npy文件，显示其中的均值和方差分别是[  9.994418 -60.24845 ]，不知道是不是因为这两个数值不对导致了结果有所偏差。

另外，我们又对fan数据集进行了测试，其中npy文件的均值和方差分别是[ 22.920975 -66.271965]，同时我们又观察到了偏高的best hmean
       ========= Test Machine Type: fan =========
       ============== TEST DATASET GENERATOR ==============
       ============== TRAIN DATASET GENERATOR ==============
       => Recorded best hmean: 0.6583
       => Detection on dev test set
       100%|██████████| 600/600 [00:42<00:00, 14.14it/s]
      => Best metric: G_z_cos_sum; Best hmean: 0.6724

**Comment by jianganbai** [comment]
更正一下，归一化方式实际为MinMax归一化为[0,1]，而非减均值除标准差的归一化。

2种机器的归一化系数都存在差异。bearing的归一化系数为[9.994418, -57.940884]，fan的归一化系数为[22.920979, -64.80351]。

test.py已更新，原代码中若不先运行train.py就运行test.py时，会缺少归一化系数文件，之前的问题可能出在这里。

可以考虑更新代码，或先运行train.py获得新归一化系数，再比较是否一致。

若不一致，应该是本地数据集的问题。
